src/WebJobs.Extensions.DurableTask/ContextImplementations/DurableOrchestrationContext.cs

```diff
@@ -1009,25 +1026,40 @@ private void ThrowIfInvalidAccess()
             }
         }
 
-        internal void SendEntityMessage(OrchestrationInstance target, string eventName, object eventContent)
+        internal void SendEntityMessage(OrchestrationInstance target, object eventContent)
         {
+            string eventName;
+
             if (eventContent is RequestMessage requestMessage)
             {
-                this.MessageSorter.LabelOutgoingMessage(
-                    requestMessage,
-                    target.InstanceId,
-                    this.InnerContext.CurrentUtcDateTime,
-                    TimeSpan.FromMinutes(this.Config.Options.EntityMessageReorderWindowInMinutes));
+                if (requestMessage.ScheduledTime.HasValue)
+                {
+                    eventName = $"op@{requestMessage.ScheduledTime.Value:o}";
```

Comment:
It looks like this event name generation logic has been duplicated into at least three different files (DurableClient.cs, DurableEntityContext.cs, and DurableOrchestrationContext.cs). Is there any way we can consolidate it into some common method?

Reply:
Yes. From a code maintenance perspective it probably makes sense to create a static class that consolidates all the event naming logic in  a single place.